### DISCUSSION TITLE
How to avoid running benchmark in debug mode?

### DISCUSSION THREAD

**Discussion by ViggoC** [facebookincubator/velox#4194] [Q&A] [answered]
I built the benchmarks by `make benchmarks-basic-build` and run one of them. The benchmark result shows like: 
```
>> ./velox_functions_prestosql_benchmarks_date_time
WARNING: Benchmark running in DEBUG mode
============================================================================
../../velox/functions/prestosql/benchmarks/DateTimeBenchmark.cpprelative  time/iter  iters/s
============================================================================
truncYear                                                  129.00ms     7.75
truncMonth                                                 121.86ms     8.21
truncDay                                                   124.44ms     8.04
truncHour                                                  125.90ms     7.94
truncMinute                                                126.39ms     7.91
truncSecond                                                  3.90ms   256.53
year                                                        51.92ms    19.26
month                                                       52.00ms    19.23
day                                                         52.36ms    19.10
hour                                                        52.68ms    18.98
hour_vector                                      103.14%    51.08ms    19.58
minute                                                      53.05ms    18.85
second                                                      51.33ms    19.48
WARNING: Benchmark running in DEBUG mode
```
I check the MakeFile, the `DCMAKE_BUILD_TYPE` is `Release`. Why there is a `debug mode warning`? Does this affect the accuracy of Benchmark?

And The performance of the time functions does not look good, `gmtime_r` and `timegm` functions is slow, is there any plan to optimize the time functions?

**Comment by lingbin** [comment]
How is your `folly` library installed? I guess there is a high probability that your `folly` library is compiled and installed in debug mode.

**Reply by ViggoC** [reply]
My environment is Debian, so I install folly like in `setup-ubuntu.sh`. 
```
function install_folly {
  github_checkout facebook/folly "${FB_OS_VERSION}"
  cmake_install -DBUILD_TESTS=OFF
}
```

**Reply by zhouyuan** [reply, marked as answer]
> My environment is Debian, so I install folly like in `setup-ubuntu.sh`.
> 
> ```
> function install_folly {
>   github_checkout facebook/folly "${FB_OS_VERSION}"
>   cmake_install -DBUILD_TESTS=OFF
> }
> ```

you may need to manually set cmake_build_type=Release in the `cmake_install` function
https://github.com/facebook/folly/blob/main/folly/Benchmark.cpp#L641

**Reply by ViggoC** [reply]
It works, thanks a lot.
